src/client/client.ts

```diff
@@ -331,6 +332,8 @@ export class GraphQLTransportWSClient {
           break;
 
         case MessageTypes.GQL_CONNECTION_KEEP_ALIVE:
+          this.wasKeepAliveReceived = true;
+          setTimeout(this.checkForTimeout, this.wsTimeout);
```

Comment:
you should save the timer id for this setTimeout, so every time you will call it after the first time you will cancel the older first.